executor/index_merge_reader.go

```diff
@@ -1107,38 +1107,41 @@ type indexMergeTableScanWorker struct {
 	memTracker *memory.Tracker
 }
 
-func (w *indexMergeTableScanWorker) pickAndExecTask(ctx context.Context) (task *indexMergeTableTask) {
+func (w *indexMergeTableScanWorker) pickAndExecTask(ctx context.Context, task **indexMergeTableTask) {
 	var ok bool
 	for {
 		waitStart := time.Now()
 		select {
-		case task, ok = <-w.workCh:
+		case *task, ok = <-w.workCh:
```

Comment:
It looks like the main point of this PR is to assign to the `task` here, so that even if we panic in `executeTask`, `startIndexMergeTableScanWorker` and `handlePickAndExecTaskPanic` could also get the `doneCh`.

I think it's not easy to realize this. I suggest we add some comments to clarify it.

Reply:
OK, I've add some comments to explain why need use the address of `task` here.